torch/csrc/jit/pickler.cpp

```diff
@@ -233,13 +233,13 @@ void Pickler::pushStorageOfTensor(const at::Tensor& tensor) {
   // typename
   pushString("storage");
   // data_type
-  std::stringstream data_type;
-  data_type << toString(tensor.scalar_type()) << "Storage";
-  pushGlobal("torch", data_type.str());
+  std::string data_type =
+    std::string(toString(tensor.scalar_type())).append("Storage");
```

Comment:
Is there any difference between `append` and `+`?

Reply:
append() is going to be slightly more efficient because it assumes we can clobber the lhs string, unlike operator+() which maintains the input lhs and rhs args.

(operator+() unconditionally copies both lhs and rhs into a newly-created string. append() just needs to ensure the buffer is big enough and then copies in the rhs. And for small strings like these, if the string size is within the default in-place buffer, there's a chance no resizing is needed)